src/stubs.ts

```diff
@@ -62,24 +59,34 @@ export function stubComponents(stubs: Record<any, any>) {
       return args
     }
 
-    if (isComponent(args)) {
-      const name = args[0]['name']
+    if (isComponent(args) || isFunctionalComponent(args)) {
+      const [type, props, children, patchFlag, dynamicProps] = args
+      const name = type['name'] || type['displayName']
       if (!name) {
         return args
       }
 
       const stub = resolveComponentStubByName(name, stubs)
 
       // we return a stub by matching Vue's `h` function
-      // where the signature is h(Component, props)
+      // where the signature is h(Component, props, slots)
       // case 1: default stub
       if (stub === true) {
-        return [createStub({ name }), {}]
+        // @ts-ignore
+        const propsDeclaration = type?.props || {}
+        return [
+          createStub({ name, props: propsDeclaration }),
+          props,
+          {},
+          patchFlag,
+          dynamicProps
+        ]
       }
 
       // case 2: custom implementation
       if (typeof stub === 'object') {
-        return [stubs[name], {}]
+        // pass the props and children, for advanced stubbing
+        return [stubs[name], props, children, patchFlag, dynamicProps]
```

Comment:
nice! I like the fact we can keep these advanced tools without increasing complexity.